smartsim/_core/launcher/step/step.py

```diff
@@ -75,7 +75,9 @@ def get_step_file(
 
     def get_colocated_launch_script(self) -> str:
         # prep step for colocated launch if specifed in run settings
-        script_path = self.get_step_file(script_name=".colocated_launcher.sh")
+        script_path = self.get_step_file(
+            script_name=f".colocated_launcher_{self.entity_name}.sh"
```

Comment:
I'm not sure that we enforce unique names for constructors ``experiment.create_model()``.  Is this potentially an edge case that we will hit?  Since this is internal only, would we want to add a very short uuid to the end that guarantees no overwrites or races?

Reply:
The issue with UUID is that if the user runs the experiment N times, we end up creating N scripts. And a similar issue arises if we try to issue an error on duplicated entity names: re-running the same entity, we might misinterpret its own launcher script for that of another entity. I think we should open a ticket about making sure two entities with the same name can not be run at the same time - from what I see in the code base, if that happens, there are other parts of the `JobManager` which break. Thoughts?

Reply:
I opened a dedicated ticket

Reply:
`get_step_file` automatically prepends the entity name to the file name. If we had to do some sort of special character removal (or any processing of the entity name), using this formatting string would make us inconsistent or have to repeat that code.

Is there any reason we don't do:

```
script_path = self.get_step_file("colocated_launcher")
```
and let `get_step_file` just spit out the standard `{entity_name}.{file_name}.{file_ext}`?